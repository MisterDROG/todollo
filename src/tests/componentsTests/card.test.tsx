import Card from "../../components/card/card";
import { fireEvent, screen } from '@testing-library/react'
import { renderWithProviders } from '../testUtils/testFunctions';

describe('Card tests', () => {

    const testingProps = {
        "branch": "+work",
        "date": "11.03.22",
        "id": "4",
        "status": "Done",
        "task": "Write program",
        "order": 1
    }

    const testingProps2 = {
        "branch": "+work",
        "date": "11.03.22",
        "id": "4",
        "status": "unDone",
        "task": "Write program",
        "order": 1
    }

    test('Is cards renders right', () => {
        const { container } = renderWithProviders(<Card todo={testingProps} />)
        const pTask = screen.getByText(/Write program/i)

        expect(pTask).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })

    test('Done button test', () => {
        renderWithProviders(<Card todo={testingProps} />)

        const buttonDone = screen.getByText('Done')
        expect(buttonDone).toBeInTheDocument()

        buttonDone && fireEvent.click(buttonDone)

        renderWithProviders(<Card todo={testingProps2} />)

        const buttonUndone = screen.getByText('unDone')
        expect(buttonUndone).toBeInTheDocument()
    })
})