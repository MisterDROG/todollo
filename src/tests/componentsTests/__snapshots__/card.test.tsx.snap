// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Is cards renders 1`] = `
<div>
  <div
    class="card"
    draggable="true"
    style="margin-top: 10px; margin-bottom: 10px;"
  >
    <div
      class="card__header"
    >
      <p
        class="card__text"
      >
        Write program
      </p>
      <button
        class="card__button-delete"
      >
        X
      </button>
    </div>
    <div
      class="card__info"
    >
      <img
        alt="clockDate image"
        class="card__logo-img"
        src="clockDateImage.png"
      />
      <p
        class="card__date"
      >
        11.03.22
      </p>
    </div>
    <button
      class="card__button-status_undone"
    >
      unDone
    </button>
  </div>
</div>
`;
